### PR TITLE
Djrenren/courses ui

### PR DESCRIPTION
**Pull request by djrenren**
This implements the Course List -> Course/Lesson List -> Lesson Heirarchy. Still waiting on @mwitte98  for a way to create lessons inside a course, but the rest is up for review.

### CODE REVIEW

File: client/src/components/Home.tsx [hunk 1]
```diff
@@ -1,54 +1,50 @@
...
 import {LessonSummary} from '../reducers/lessonSummaryReducer';
+import {Course} from '../reducers/courseReducer';
 import Page from './util/Page';
 import * as React from 'react';
 
 export interface HomeProps {
+    courses?: Course[];
     lessons?: LessonSummary[];
     loading?: boolean;
     onCreateLessonClick?: () => void;
```
Comment (mwitte98) [inline client/src/components/Home.tsx]: Do we still need lessons, LessonSummary, and onCreateLessonClick in the home component?

Comment (djrenren) [reply client/src/components/Home.tsx]: Nope

Comment (mwitte98) [reply client/src/components/Home.tsx]: Can you remove them? Looks good after that

File: client/src/containers/CourseContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        }
+    };
+}
+
+const LessonCreatorContainer = connect(
+    mapStateToProps,
+    mapDispatchToProps
+)(CourseComp);
+
+export default LessonCreatorContainer;
```
Comment (mwitte98) [inline client/src/containers/CourseContainer.tsx]: Also rename this to CourseContainer

**Review by mwitte98 — CHANGES_REQUESTED** [review]

**Review by mwitte98 — APPROVED** [review]

**Pull request merged**